_stbt/core.py

```diff
@@ -1622,10 +1637,14 @@ def teardown(self):
             source = None
         if not self.novideo:
```

Comment:
So that was back when we only had a single pipeline and I don't think it applies any more, in particular since after this PR we are no longer depending on EOS to exit the mainloop.